### PR TITLE
Implement "Add unambiguous imports on the fly" option

### PR DESCRIPTION
**Pull request by dima74**
changelog: Implement "Add unambiguous imports on the fly" option. You can turn it on in `Settings | Editor | General | Auto Import`. It can be useful e.g. when copying some code from a web site.

Example:

Option in settings:

---

I united `AutoImportFix` and `AutoImportHintFix` in order to get simpler code, and also in other languages there is no such division. Also now `candidates` are stored in `AutoImportFix`

### CODE REVIEW

**Comment by Kobzol** [comment]
Pretty cool! I suppose that this supersedes https://github.com/intellij-rust/intellij-rust/pull/7597?

**Comment by dima74** [comment]
@Kobzol I think #7597 is needed too, because "unambiguous imports on the fly" option will be disabled by default (as in Java/Kotlin). Also even if user enable "unambiguous ..." option, #7597 will be useful when copying code with ambiguous name

**Review by Undin — COMMENTED** [review]
Looks good in general but comments

File: src/main/kotlin/org/rust/ide/inspections/import/RsReferenceImporter.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+package org.rust.ide.inspections.import
+
+import com.intellij.codeInsight.daemon.ReferenceImporter
+import com.intellij.openapi.editor.Editor
+import com.intellij.psi.PsiFile
+import org.rust.ide.settings.RsCodeInsightSettings
+import org.rust.lang.core.psi.RsFile
+
+class RsReferenceImporter : ReferenceImporter {
+    override fun autoImportReferenceAtCursor(editor: Editor, file: PsiFile): Boolean = false
```
[the remark this answers is not in the document]
Comment (dima74) [reply src/main/kotlin/org/rust/ide/inspections/import/RsReferenceImporter.kt:15]: No such plans at the moment, but in general it is possible. Could you please advise, what is wrong with help comment?
image

[the remark this answers is not in the document]
Comment (Undin) [reply src/main/kotlin/org/rust/ide/inspections/import/RsReferenceImporter.kt:15]: > Could you please advise, what is wrong with help comment?

Oh, nothing wrong. Sorry, my mistake, everything is fine

**Review by Undin — APPROVED** [review]
bors r+

**Pull request merged**